go-controller/pkg/clustermanager/userdefinednetwork/controller.go

```diff
@@ -410,6 +415,10 @@ func (c *Controller) syncUserDefinedNetwork(udn *userdefinednetworkv1.UserDefine
 				return nil, fmt.Errorf("failed to delete NetworkAttachmentDefinition [%s/%s]: %w", udn.Namespace, udn.Name, err)
 			}
 
+			if c.networkManager.GetControllerNetwork(util.GenerateUDNNetworkName(udn.Namespace, udn.Name)) != nil {
```

Comment:
Do you think it would do good to have comments here. This can be a hotspot where people might look at it an ask "why are we waiting?"

Reply:
Great idea, will do.